xarray/core/variable.py

```diff
@@ -893,16 +893,27 @@ def reduce(self, func, dim=None, axis=None, keep_attrs=False,
         if dim is not None and axis is not None:
             raise ValueError("cannot supply both 'axis' and 'dim' arguments")
 
+        if 'cum' in func.__name__:
```

Comment:
Can we put these error checks in `ops.cumsum` instead? It's poor separation of concerns to do these checks over here.